### ISSUE TITLE
Extension methods shouldn't be suggested within class body

### ISSUE THREAD

**Issue by parlough** [dart-lang/sdk#54746]
In a class body, extension methods are included in the suggestions. Just like normal methods, I think it probably makes sense to not suggest this here. It's common to come across this now since the SDK has some extensions added on `Object`.

```dart
class Test {
  r^
}

extension on Test {
  int random() {
    return 3;
  }
}
```

suggests:

```dart
class Test {
  random()
}

extension on Test {
  int random() {
    return 3;
  }
}
```

Commit tested at: 55310a1c7028e495ec9084beee05d4e25f7eaca8

**Comment by bwilkerson** [comment]
Yes, we should not be suggesting a method (extension or otherwise) in the body of a class-like declaration.

**Comment by scheglov** [comment]
https://dart-review.googlesource.com/c/sdk/+/355510
